source/speech/speech.py

```diff
@@ -1374,8 +1387,16 @@ def speakTypedCharacters(ch: str):
 		clearTypedWordBuffer()
 		if log.isEnabledFor(log.IO):
 			log.io("typed word: %s" % typedWord)
-		if config.conf["keyboard"]["speakTypedWords"] and not typingIsProtected:
-			speakText(typedWord)
+		typingEchoMode = config.conf["keyboard"]["speakTypedWords"]
+		if typingEchoMode > TypingEcho.OFF.value and not typingIsProtected:
```

Comment:
Could you be consistent with what is done later for characters, i.e. use `!=` instead of `>`? I also feel it's slightly easier to read/understand since with this syntax, you do not need to know the numeric values of each `TypingEcho` member.

```suggestion
		if typingEchoMode != TypingEcho.OFF.value and not typingIsProtected:
```